File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/folder/FolderCreationEvent.kt
package dev.subfly.yabacore.state.folder

import dev.subfly.yabacore.model.utils.YabaColor

sealed class FolderCreationEvent {
    data class OnInitWithFolder(val folderIdString: String?) : FolderCreationEvent()
    data class OnSelectNewParent(val newParentId: String?) : FolderCreationEvent()
    data class OnSelectNewColor(val newColor: YabaColor) : FolderCreationEvent()
    data class OnSelectNewIcon(val newIcon: String) : FolderCreationEvent()
    data class OnChangeLabel(val newLabel: String) : FolderCreationEvent()
    data class OnChangeDescription(val newDescription: String) : FolderCreationEvent()
    data object OnSave: FolderCreationEvent()
}

File: Darwin/YABACore/State/Creation/Folder/FolderCreationEvent.swift
//
//  FolderCreationEvent.swift
//  YABACore
//
//  Parity with Compose `FolderCreationEvent`.
//

import Foundation

public enum FolderCreationEvent: Sendable {
    case onInitWithFolder(folderId: String?)
    case onSelectNewParent(parentFolderId: String?)
    case onSelectNewColor(YabaColor)
    case onSelectNewIcon(String)
    case onChangeLabel(String)
    case onChangeDescription(String)
    case onSave

    /// One-shot create (legacy helper).
    case create(folderId: String, label: String, description: String?, parentFolderId: String?)
}
